lib/prom_ex/plugins/plug_router.ex

```diff
@@ -0,0 +1,248 @@
+     `plug_router_metric_prefix` is `{otp_app}_prom_ex_plug_router`.
+    - `ignore_routes`: This option is OPTIONAL and is used to ignore certain paths.
+
+    To use plugin in your application, add the following to your PromEx module:
+
+    ```
+    defmodule WebApp.PromEx do
+      use PromEx, otp_app: :web_app
+
+      alias PromEx.Plugins
+
+      @impl true
+      def plugins do
+        [
+          ...
+          {Plugins.PlugRouter,
+            event_prefix: [:webapp, :router], metric_prefix: [:prom_ex, :router], routers: [WebApp.Router]},
+          {Plugins.PromEx, metric_prefix: [:prom_ex, :prom_ex]}
+        ]
+      end
+
+      @impl true
+      def dashboard_assigns do
+        [
+          datasource_id: "...",
+          plug_router_metric_prefix: "prom_ex_router",
+          prom_ex_metric_prefix: "prom_ex_prom_ex"
+        ]
+      end
+
+      @impl true
+      def dashboards do
+        [
+          ...
+          {:prom_ex, "plug_router.json"}
+        ]
+      end
+    end
+    ```
+
+    To ignore certain paths, pass a list of routes using the `:ignore_routes` option
+
+    ```
+    defmodule WebApp.PromEx do
+      use PromEx, otp_app: :web_app
+
+      @impl true
+      def plugins do
+        [
+          ...
+          {PromEx.Plugins.PlugRouter,
+            event_prefix: [:webapp, :router], metric_prefix: [:prom_ex, :router], routers: [WebApp.Router],
+            ignore_routes: ["/metrics"]}
+        ]
+      end
+
+      @impl true
+      def dashboards do
+        [
+          ...
+          {:prom_ex, "plug_router.json"}
+        ]
+      end
+    end
+    ```
+    """
+
+    use PromEx.Plugin
+    require Logger
+    alias Plug.Conn
+
+    @stop_event [:prom_ex, :router, :stop]
+    @default_event_prefix [:plug, :router_dispatch]
+
+    @impl true
+    def event_metrics(opts) do
+      otp_app = Keyword.fetch!(opts, :otp_app)
+      metric_prefix = Keyword.get(opts, :metric_prefix, PromEx.metric_prefix(otp_app, :plug_router))
+      event_prefix = Keyword.fetch!(opts, :event_prefix)
+      set_up_telemetry_proxy(event_prefix)
+
+      [
+        http_events(metric_prefix, opts)
+      ]
+    end
+
+    defp set_up_telemetry_proxy(event_prefix) do
+      :telemetry.attach(
+        {__MODULE__, :stop},
+        event_prefix ++ [:stop],
+        &__MODULE__.handle_proxy_router_event/4,
+        %{}
+      )
+
+      :telemetry.attach(
+        {__MODULE__, :exception},
+        @default_event_prefix ++ [:exception],
+        &__MODULE__.handle_proxy_router_event/4,
+        %{}
+      )
+    end
+
+    def handle_proxy_router_event(_, measurements, meta, _) do
+      :telemetry.execute(@stop_event, measurements, meta)
+    end
+
+    defp http_events(metric_prefix, opts) do
+      http_metrics_tags = [:status, :method, :path]
+
+      routers =
+        opts
+        |> Keyword.fetch!(:routers)
+        |> MapSet.new()
+
+      ignore_routes =
```

Comment:
I haven't looked at the telemetry event metadata so this is more of a question. Is this the going to be compared to a normalized route (something like `/users:id`), or is it the raw route (something like `/users/1234567890`).

If the route is normalized, I think string comparison via a MapSet works great. If it is not normalized, we'll also probably want to have a RegEx option so that the user can capture all those routes.